packages/daf-web-did/src/identity-controller.ts

```diff
@@ -0,0 +1,55 @@
+import { AbstractIdentityController, AbstractKeyManagementSystem, AbstractIdentityStore } from 'daf-core'
+import Debug from 'debug'
+const debug = Debug('daf:web-did:identity-controller')
+
+export class IdentityController extends AbstractIdentityController {
+  private did: string
+  private kms: AbstractKeyManagementSystem
+  private identityStore: AbstractIdentityStore
+
+  constructor(options: {
+    did: string
+    kms: AbstractKeyManagementSystem
+    identityStore: AbstractIdentityStore
+  }) {
+    super()
+    this.did = options.did
+    this.kms = options.kms
+    this.identityStore = options.identityStore
+  }
+
+  async addService(service: { id: string; type: string; serviceEndpoint: string }): Promise<any> {
+    return Promise.reject('[daf-web-did identity-controller] addService not implemented')
+  }
+
+  async addPublicKey(type: 'Ed25519' | 'Secp256k1', proofPurpose?: string[]): Promise<any> {
```

Comment:
This name is a bit misleading since it's also creating the key to be added and there's no way to add your own key.

We probably need to change the abstract types to deal with this naming issue. Please create a separate issue if that's the case, so that it's not lost.

Reply:
I created an issue for this: https://github.com/uport-project/daf/issues/171